app/models/concerns/audit_extensions.rb

```diff
@@ -93,15 +86,32 @@ def known_auditable_types
       end
 
       def taxed_only_by_location
-        locations = Location.current || Location.my_locations.reorder(nil)
-        by_auditable_types(location_taxable).
-          where(id: TaxableTaxonomy.where(taxonomy: locations, taxable_type: 'Audited::Audit').select(:taxable_id))
+        join_locations.where(arel_taxed_only_by_location)
       end
 
       def taxed_only_by_organization
-        organizations = Organization.current || Organization.my_organizations.reorder(nil)
-        by_auditable_types(organization_taxable).
-          where(id: TaxableTaxonomy.where(taxonomy: organizations, taxable_type: 'Audited::Audit').select(:taxable_id))
+        join_organizations.where(arel_taxed_only_by_organization)
+      end
+
+      def taxed_and_untaxed
+        in_taxonomy_scope do |nodes|
```

Comment:
This is very unclear method it sounds it's doing the same as `with_taxonomy_scope`, the purpose of the block is not very clear on the first sight (without actually reading the `in_taxonomy_scope` body).
If we could get rid of the `unless` branches from `in_taxonomy_scope`, the method would not be needed and we could just do:
```ruby
statements = # code in the block body
taxonomy_join_scope.where(statements)
```

Reply:
Yeah, the purpose of in_taxonomy_scope is to replace the with_taxonomy_scope for more efficient query. I think it might be good to replace with_taxonomy_scope in the future

Reply:
I think block is still needed so that the arel query can be customized better and more flexible, such as the grouping.

Reply:
@ezr-ondrej as per discussed, I removed the "in_taxonomy_scope" and will make it reusable for other places in a separate commit.